src/features/notification/handler.ts

```diff
@@ -11,25 +11,100 @@
  */
 import * as Hapi from '@hapi/hapi'
 import * as Joi from 'joi'
-import { sendSMS } from './service'
+import {
+  SMSTemplateType,
+  sendSMSClickatell,
+  sendSMSInfobip
+} from './sms-service'
+import {
+  EmailTemplateType,
+  TemplateVariables,
+  sendEmail
+} from './email-service'
+import { SMS_PROVIDER } from './constant'
+import { APPLICATION_CONFIG_URL } from '@countryconfig/constants'
+import fetch from 'node-fetch'
+import { URL } from 'url'
+import { logger } from '@countryconfig/logger'
 
-interface INotificationPayload {
-  msisdn: string
-  message: string
+type NotificationPayload = {
+  type: 'email' | 'sms'
+  templateName: EmailTemplateType | SMSTemplateType
+  recipient: string
+  locale: string
+  variables: TemplateVariables
   convertUnicode?: boolean
 }
 
+interface ILoginBackground {
+  backgroundColor: string
+  backgroundImage: string
+  imageFit: string
+}
+interface ICountryLogo {
+  fileName: string
+  file: string
+}
+interface IApplicationConfig {
+  APPLICATION_NAME: string
+  COUNTRY: string
+  COUNTRY_LOGO: ICountryLogo
+  SENTRY: string
+  LOGROCKET: string
+  LOGIN_BACKGROUND: ILoginBackground
+}
+interface IApplicationConfigResponse {
+  config: IApplicationConfig
+}
+
 export const notificationScheme = Joi.object({
-  msisdn: Joi.string(),
-  message: Joi.string(),
-  convertUnicode: Joi.boolean()
-})
+  type: Joi.string().valid('sms', 'email')
+}).unknown(true)
+
+export async function notificationHandler(request: Hapi.Request) {
```

Comment:
This handler doesn't return any response currently. This is a tech debt I guess.